datacats/error.py

```diff
@@ -4,7 +4,10 @@
 class DatacatsError(Exception):
 
     def __init__(self, message, format_args=(), parent_exception=None):
-        self.message = message
+        try:
+            self.message = message.format(*format_args)
+        except (ValueError, KeyError):
```

Comment:
I agree. This was the original format for DatacatsError and it is really obscure and error prone. Not only parameters are passed to the exception, they are not named. So you have to count "{0}, {1}" each time and it is really obscure.